starboard/nplb/posix_compliance/posix_epoll_test.cc

```diff
@@ -0,0 +1,279 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <sys/epoll.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "starboard/common/socket.h"
+#include "starboard/configuration_constants.h"
+#include "starboard/nplb/posix_compliance/posix_socket_helpers.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace starboard {
+namespace nplb {
+namespace {
+
+constexpr int kMaxEvents = 10;
+constexpr int kMaxBuf = 1024;
+
+class PosixEpollTest : public testing::Test {
+ public:
+  void SetUp() override {
+    // Create epoll object, arg doesn't matter as long as it is > 0.
+    epfd = epoll_create(1);
+    ASSERT_GT(epfd, 0) << strerror(errno);
+
+    // Create non-blocking socket for use in all tests.
+    socket_fd = ::socket(AF_INET, SOCK_STREAM | SOCK_NONBLOCK, IPPROTO_TCP);
+    ASSERT_GE(socket_fd, 0) << strerror(errno);
+
+    // Set epoll_event data.
+    epev.data.fd = socket_fd;
+    epev.events = EPOLLIN | EPOLLOUT;
+
+    // Get socket for use in tests similar to socket_helper.cc.
+    ASSERT_TRUE(
+        PosixGetLocalAddressIPv4(reinterpret_cast<sockaddr*>(&address)) == 0 ||
+        PosixGetLocalAddressIPv6(reinterpret_cast<sockaddr*>(&address)) == 0);
+    address.sin6_port = htons(PosixGetPortNumberForTests());
+  }
+
+  void TearDown() override {
+    EXPECT_EQ(close(socket_fd), 0) << strerror(errno);
+    EXPECT_EQ(close(epfd), 0) << strerror(errno);
+  }
+
+ protected:
+  // TODO(b/412696447): Correct member variable style.
+
+  int epfd;
+  int socket_fd;
+  epoll_event epev;
```

Comment:
Since `epev` is used everywhere, could we give it a better name, like `polling_event`, or just `event`? As it is, is a shortening but doesnt' even respect the word order (should be `ep_ev` in any case). But since it's pervasive and it's meant to be readable, please correct.